plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/metastore/glue/TestDeltaS3AndGlueMetastoreTest.java

```diff
@@ -61,7 +62,7 @@ protected void validateDataFiles(String partitionColumn, String tableName, Strin
         {
             String locationDirectory = location.endsWith("/") ? location : location + "/";
             String partitionPart = partitionColumn.isEmpty() ? "" : partitionColumn + "=[a-z0-9]+/";
-            assertThat(dataFile).matches("^" + locationDirectory + partitionPart + "[a-zA-Z0-9_-]+$");
+            assertThat(dataFile).matches("^" + Pattern.quote(locationDirectory) + partitionPart + "[a-zA-Z0-9_-]+$");
```

Comment:
why not quote the `partitionPart` too?

Reply:
because some tests failed when I did quote it

Reply:
oh indeed, here partitionPart is a pattern, not a value